src/scene/forward-renderer.js

```diff
@@ -823,6 +828,143 @@ class ForwardRenderer {
         }
     }
 
+    // placeholder LUT textures for area light
+    _createAreaLightPlaceholderLuts() {
+        var placeholderLutTex =  new Texture(this.device, {
+            width: 2,
+            height: 2,
+            format: PIXELFORMAT_R8_G8_B8_A8
+        });
+        placeholderLutTex.name = 'placeholder';
+
+        var pixels = placeholderLutTex.lock();
+        for (var i = 0; i < 4; i++) {
+            for (var c = 0; c < 4; c++) {
+                pixels[i * 4 + c] = 0;
+            }
+        }
+        placeholderLutTex.unlock();
+
+        this.device.scope.resolve('areaLightsLutTex1').setValue(placeholderLutTex);
+        this.device.scope.resolve('areaLightsLutTex2').setValue(placeholderLutTex);
+    }
+
+    // creates LUT texture used by area lights
+    _uploadAreaLightLuts(resource) {
+
+        var versions = new Int16Array(resource, 32768 * 4, 2);
+
+        this._areaLightLuts = {
+            data1: new Float32Array(resource, 0, 16384),
+            data2: new Float32Array(resource, 16384 * 4, 16384),
+            majorVersion: versions[0],
+            minorVersion: versions[1]
+        };
```

Comment:
one more comment .. is there a need to store these luts here at all on this? When they get uploaded to textures, there is no need to keep cp copy I assume?

Reply:
hmm - these are views of the asset resource - so it isn't a copy - but probably would prevent the any cleanup of the asset resource - which would be nice to save CPU memory - I will remove this.

Reply:
new Float32Array creates a copy of the array

Reply:
fixed